### PR TITLE
139 improve nmdl list speed

### PR DESCRIPTION
**Pull request by cmwill**

### CODE REVIEW

**Review by marshall-sg — CHANGES_REQUESTED** [review]
A couple minor changes are requested.  Let me know if there are any questions.

Otherwise it looks good and has resulted in a dramatic increase in speed and usability for large repositories (e.g., several minutes down to several seconds, so about an order of magnitude improvement).

File: datalake/common/handlers/Git.cpp [hunk 1]
```diff
@@ -104,25 +104,64 @@ namespace netmeld::datalake::handlers {
...
+    topLevel[topLevel.size() - 1] = '/';
+    oss.str("");
+
+    oss << "git log --pretty=format:\"%B\"";
     std::istringstream iss(nmcu::cmdExecOut(oss.str()));
 
+
+    std::vector<std::string> logs;
+    logs.reserve(vde.size() * 3);
+
```
[the remark this answers is not in the document]
Comment (cmwill) [reply datalake/common/handlers/Git.cpp:123]: Yeah, I used it at one point and forgot to remove it when I was reworking my code.

File: datalake/nmdl-list/nmdl-list.cpp [hunk 2]
```diff
@@ -164,7 +164,8 @@ class Tool : public nmdlt::AbstractDatalakeTool
     {
       const auto& dataLake     {getDatalakeHandler()};
       const auto& time         {opts.getValueAs<nmco::Time>("before")};
-      const auto& dataEntries  {dataLake->getDataEntries(time)};
+      const auto& dataEntries  {dataLake->getDataEntries(time,
+        opts.exists("ingest-script") || opts.exists("by-tool"))};
```
[the remark this answers is not in the document]
Comment (cmwill) [reply datalake/nmdl-list/nmdl-list.cpp:168]: Oh, I've been misunderstanding how unbinned data is displayed. I get it now, I'll go ahead and add that check.

File: datalake/common/handlers/Git.cpp [hunk 3]
```diff
@@ -104,25 +104,64 @@ namespace netmeld::datalake::handlers {
...
+    std::unordered_map<std::string, std::pair<std::string, std::string>> parsedData;
+    parsedData.reserve(vde.size() * 2);
```
[the remark this answers is not in the document]
Comment (cmwill) [reply datalake/common/handlers/Git.cpp:121]: Multiplying by 2 was due to me trying to minimize collisions and possibly misunderstanding how unordered_maps work. If preallocation will cause issues with large repos, it'll probably be better for me to remove the `reserve` call and let the map handle it on its own.

**Review by marshall-sg — APPROVED** [review]
Looks good enough.  Thanks!

**Pull request merged**